pkg/paravirt/paravirt_protected_entity_type_manager.go

```diff
@@ -282,26 +252,140 @@ func (this *ParaVirtProtectedEntityTypeManager) CreateFromMetadata(ctx context.C
 		return nil, fmt.Errorf("CloneFromSnapshot is canceled: %s/%s in the Supervisor Cluster", svcClone.Namespace, svcClone.Name)
 	}
 	this.logger.Infof("CreateFromMetadata: CloneFromSnapshot %s/%s is completed in Supervisor Cluster. Phase: %v", svcClone.Namespace, svcClone.Name, svcClone.Status.Phase)
-
 	// Get a fresh Supervisor PVC object
 	svcPvcUpdated, err := this.svcKubeClientSet.CoreV1().PersistentVolumeClaims(this.svcNamespace).Get(context.TODO(), svcPVC.Name, metav1.GetOptions{})
 	if err != nil {
 		this.logger.Errorf("Failed to get PVC %s/%s from Supervisor Cluster: %v", this.svcNamespace, svcPVC.Name, err)
 		return nil, errors.Wrapf(err, "Failed to get PVC from Supervisor Cluster")
 	}
-	svcPVC = *svcPvcUpdated
+	svcPVC = svcPvcUpdated
 
-	// Create a PV in Guest Cluster that points to PVC Name in Supervisor Cluster (static provisioning)
-	namespace := cloneFromSnapshotNamespace
-	pvcName := svcPVC.Name
-	pvUUID, err := uuid.NewRandom()
+	gcPV, err := this.createGuestPV(gcPVCNamespace, gcPVCName, svcPVC, CSIDriverName)
+	if err != nil {
+		return nil, err
+	}
+
+	gcPVC, err := this.createGuestPVC(gcPVCNamespace, gcPVCName, gcPV.Name, svcPVC)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Could not generate PV name in the Guest Cluster for Supervisor Cluster PVC %s/%s",
+		return nil, err
+	}
+
+	err = this.updateCloneFromSnapshotStatus(cloneFromSnapshotNamespace, cloneFromSnapshotName, svcClone)
```

Comment:
minor comment here. what about the following style for the error checking? It is more compact and prevents the error from being spread outside the if block. 

```
if err := this.updateCloneFromSnapshotStatus(cloneFromSnapshotNamespace, cloneFromSnapshotName, svcClone); err != nil {
    ...
}